ostelco-ios-client/src/features/splash/SplashViewController.swift

```diff
@@ -26,208 +26,155 @@ class SplashViewController: UIViewController {
+            print("handleContext")
+            UserManager.sharedInstance.user = context.customer
+            if let region = context.getRegion() {
+                OnBoardingManager.sharedInstance.region = region
+                let segueIdentifier = getSegueFromRegionResponse(region: region)
+                print("handleContext switch to segue \(segueIdentifier)")
+                performSegue(withIdentifier: segueIdentifier, sender: self)
+            } else {
+                print("handleContext switch to showCountry")
+                performSegue(withIdentifier: "showCountry", sender: self)
+            }
+        } else {
             // for 404 the context will be empty
+            print("handleContext empty context 404")
             performSegue(withIdentifier: "showSignUp", sender: self)
-            return
         }
-        UserManager.sharedInstance.user = context.customer
+        let resolved: Promise<Void?> = getFullfilledPromise()
+        return resolved
+    }
 
-        var segueIdentifier: String
-        guard let region = context.getRegion() else {
-            performSegue(withIdentifier: "showCountry", sender: self)
-            return
+    func getRejectedPromise<T>() -> Promise<T?> {
+        return Promise<T?> { seal in
+            let message = "Terminating promise chain, ignore"
+            seal.reject(createError(message))
+        }
+    }
+
+    func getFullfilledPromise<T>() -> Promise<T?> {
+        let empty:T? = nil
+        return Promise<T?> { seal in
+            seal.resolve(empty, nil)
         }
-        OnBoardingManager.sharedInstance.region = region
+    }
+
+    func getRejectedContext() -> Promise<Context?> {
+        let empty: Context? = nil
+        return Promise<Context?> { seal in
+            func rejectPromise(_ error: RequestError) {
+                let failure = "Failed to fetch user context from server: \(error.userMessage)"
+                seal.reject(createError(failure))
+            }
+            spinnerView = showSpinner(onView: view)
+            APIManager.sharedInstance.context.load().onSuccess { data in
+                if let context: Context = data.typedContent(ifNone: nil) {
+                    seal.resolve(context, nil)
+                } else {
+                    let failure = "Failed to parse user context from server response."
+                    seal.reject(createError(failure))
+                }
+                }.onFailure { error in
+                    if let statusCode = error.httpStatusCode, statusCode == 404 {
+                        // not found, assume user dosen't exist
+                        seal.resolve(empty, nil)
+                    } else {
+                        rejectPromise(error)
+                    }
+                }.onCompletion { _ in
+                    self.removeSpinner(self.spinnerView)
+            }
+        }
+    }
+
+    func getSegueFromRegionResponse(region: RegionResponse) -> String {
```

Comment:
[LoginViewController2](https://github.com/ostelco/ostelco-ios-client/blob/feature/use-promise/ostelco-ios-client/src/features/login/LoginViewController2.swift ) could also benefit from this function, it has similar logic to the splash screen, and uses the exact same logic for routing and setting credentials after login is complete

Reply:
Except for how it performs the segue: 
```
self.perform(#selector(self.showEKYCLastScreen), with: nil, afterDelay: 0.5)
```
For some reason in needs the delay, else the segue won't show, I assume this is an logic error from my part